WireDomain/Sources/WireDomain/Notifications/Builders/NewMessageNotificationBuilder.swift

```diff
@@ -0,0 +1,378 @@
+            let quotedMessageId = UUID(uuidString: textMessageData.quote.quotedMessageID)
+            let quotedMessage = await messageLocalStore.fetchMessage(
+                id: quotedMessageId,
+                conversationID: context.conversationID.uuid,
+                conversationDomain: context.conversationID.domain
+            )
+
+            isMention = await messageLocalStore.isMessageMentioningSelf(text: textMessageData)
+            isReply = await messageLocalStore.isMessageQuotingSelf(quotedMessage: quotedMessage)
+
+        } else {
+            isMention = false
+            isReply = false
+        }
+
+        let content = UNMutableNotificationContent()
+
+        let format: NotificationBody.MessageBodyFormat = if isMention {
+            .mentionedWithUnknownSender
+        } else if isReply {
+            .repliedWithUnknownSender
+        } else {
+            .sentWithUnknownSender
+        }
+
+        let body = NotificationBody.newMessage(
+            format
+        )
+
+        // No thread identifier for ephemeral messages as we only want to group non ephemeral ones.
+        content.body = body.make()
+        content.categoryIdentifier = makeCategory()
+        content.sound = makeSound()
+        content.userInfo = makeUserInfo()
+
+        return content
+    }
+
+    // MARK: - Helpers
+
+    private func makeTitle(
+    ) -> String? {
+        let isGroupConversation = context.isGroupConversation
+        let teamName = context.teamName
+        let conversationName = context.conversationName
+        let senderName = context.senderName
+
+        guard let conversationName, let senderName else {
+            return nil
+        }
+
+        let format: NotificationTitle.MessageTitleFormat = if isGroupConversation {
+            if let teamName {
+                .conversationInTeam(conversation: conversationName, team: teamName)
+            } else {
+                .conversation(conversation: conversationName)
+            }
+        } else {
+            if let teamName {
+                .senderInTeam(sender: senderName, team: teamName)
+            } else {
+                .sender(sender: senderName)
+            }
+        }
+
+        return NotificationTitle
+            .newMessage(format)
+            .make()
+    }
+
+    private func makeSound(type: NotificationSound = .default) -> UNNotificationSound {
+        let notificationSoundName = UNNotificationSoundName(type.rawValue)
+        return UNNotificationSound(named: notificationSoundName)
+    }
+
+    private func makeCategory() -> String {
+        let category = NotificationCategory.unmutedConversation
+        return category.rawValue
+    }
+
+    private func makeUserInfo() -> [AnyHashable: Any] {
+        var userInfo: [AnyHashable: Any] = [:]
+
+        userInfo["selfUserIDString"] = context.selfUserID
+        userInfo["senderIDString"] = context.senderID
+        userInfo["conversationIDString"] = context.conversationID.uuid
```

Comment:
is this going to be consumed to the main app? if so might be good to use constant for keys for safe access